stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -815,24 +828,53 @@ extension _JSONEncoder {
         }
     }
 
-    fileprivate func box<T : Encodable>(_ value: T) throws -> NSObject {
+    fileprivate func box(_ value: Encodable) throws -> NSObject {
```

Comment:
Slight nit: Do you mind keeping this method right above the implementation of `box_`? (i.e. move the dictionary version above this one)